packages/ia-topnav/src/ia-topnav.js

```diff
@@ -0,0 +1,237 @@
+      userMenuAnimate: { type: Boolean },
+      userMenuOpen: { type: Boolean },
+    };
+  }
+
+  constructor() {
+    super();
+    this.config = {};
+    this.mediaMenuAnimate = false;
+    this.mediaMenuOpen = false;
+    this.searchMenuAnimate = false;
+    this.searchMenuFade = false;
+    this.searchMenuOpen = false;
+    this.searchSubmitted = false;
+    this.signedOutMenuOpen = false;
+    this.userMenuAnimate = false;
+    this.userMenuOpen = false;
+    this.mediaSliderOpen = false;
+    this.mediaSliderAnimate = false;
+    this.selectedMenuOption = '';
+  }
+
+  updated(changedProperties) {
+    const { mediaMenuOpen, mediaSliderOpen } = this;
+    const menuClosed = changedProperties.has('mediaMenuOpen')
+      && changedProperties.get('mediaMenuOpen')
+      && !mediaMenuOpen;
+
+    if (menuClosed && mediaSliderOpen) {
+      this.mediaSliderOpen = false;
+      this.mediaSliderAnimate = false;
+      this.selectedMenuOption = '';
+    }
+  }
+
+  closeMediaSlider() {
+    this.mediaSliderAnimate = true;
+    this.mediaSliderOpen = false;
+    this.selectedMenuOption = '';
+  }
+
+  toggleMediaSlider() {
+    if (!this.mediaSliderOpen) {
+      this.mediaSliderAnimate = true;
+      this.mediaSliderOpen = !this.mediaSliderOpen;
+      this.mediaMenuOpen = true;
+    }
+  }
+
+  mediaMenu() {
+    this.userMenuOpen = false;
+    this.searchMenuOpen = false;
+    this.mediaMenuAnimate = true;
+    this.signedOutMenuOpen = false;
+    this.mediaMenuOpen = !this.mediaMenuOpen;
+  }
+
+  signedOutMenu() {
+    this.userMenuOpen = false;
+    this.searchMenuOpen = false;
+    this.mediaMenuOpen = false;
+    this.signedOutMenuOpen = !this.signedOutMenuOpen;
+  }
+
+  searchMenu() {
+    this.userMenuOpen = false;
+    this.mediaMenuOpen = false;
+    this.signedOutMenuOpen = false;
+    this.searchMenuAnimate = true;
+    this.searchMenuFade = true;
+    this.searchMenuOpen = !this.searchMenuOpen;
+    if (this.searchMenuOpen) {
+      window.setTimeout(() => {
+        this
+          .shadowRoot
+          .querySelector('primary-nav')
+          .shadowRoot
+          .querySelector('nav-search')
+          .shadowRoot
+          .querySelector('.search-field').focus();
+      }, 0);
+    }
+  }
+
+  userMenu() {
+    this.searchMenuOpen = false;
+    this.mediaMenuOpen = false;
+    this.signedOutMenuOpen = false;
+    this.userMenuAnimate = true;
+    this.userMenuOpen = !this.userMenuOpen;
+  }
+
+  closeMenus() {
+    this.mediaMenuOpen = false;
+    this.mediaSliderOpen = false;
+    this.searchMenuOpen = false;
+    this.signedOutMenuOpen = false;
+    this.userMenuOpen = false;
+    this.selectedMenuOption = '';
+  }
+
+  navSearch(e) {
+    this.searchSubmitted = true;
+    const { originalEvent, formEl } = e.detail;
+    const searchIn = document
+      .querySelector('ia-topnav')
+      .shadowRoot
+      .querySelector('search-menu')
+      .shadowRoot
+      .querySelector('[name=sin]:checked')
+      .value;
```

Comment:
This is tunneling awfully deep into other components to get a value. Can you store the checked value as a property or emit an event from the child components when that thing changes so consumers like this component can be aware of these changes?

Reply:
Yep. One of those bits of code that I put in initially while I rearranged elements and meant to go back to.